File: project-euler/euler-java/src/test/java/com/github/adrianomoreira/projecteuler100/Problem002Test.java
package com.github.adrianomoreira.projecteuler100;

import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

public class Problem002Test {

    @DisplayName("fibonacciEvenSum(X) should return Y")
    @Test
    public void test() {
        assertEquals(44, Problem002.fibonacciEvenSum(10));
        assertEquals(3382, Problem002.fibonacciEvenSum(18));
        assertEquals(60696, Problem002.fibonacciEvenSum(23));
        assertEquals(350704366, Problem002.fibonacciEvenSum(43));
    }

}

File: project-euler/euler-kotlin/src/test/kotlin/Problem002Test.kt
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test

internal class Problem002Test {

    @Test
    fun fibonacciEvenSum() {
        assertEquals(44, fibonacciEvenSum(10))
        assertEquals(3382, fibonacciEvenSum(18))
        assertEquals(60696, fibonacciEvenSum(23))
        assertEquals(350704366, fibonacciEvenSum(43))
    }

}